File: data/config-schema/src/main/java/net/corda/schema/configuration/BootConfig.java
package net.corda.schema.configuration;

import static net.corda.schema.configuration.MessagingConfig.MAX_ALLOWED_MSG_SIZE;

/**
 * Configuration paths for values used to bootstrap the worker
 */
@SuppressWarnings("unused")
public final class BootConfig {
    private BootConfig() {
    }

    public static final String INSTANCE_ID = "instanceId";
    public static final String TOPIC_PREFIX = "topicPrefix";
    public static final String BOOT_MAX_ALLOWED_MSG_SIZE = MAX_ALLOWED_MSG_SIZE;

    public static final String BOOT_KAFKA = "kafka";
    public static final String BOOT_KAFKA_COMMON = BOOT_KAFKA + ".common";
    public static final String BOOT_CRYPTO = "crypto";

    public static final String BOOT_DB = "db";
    public static final String BOOT_JDBC_URL = BOOT_DB + ".database.jdbc.url";
    public static final String BOOT_JDBC_USER = BOOT_DB + ".database.user";
    public static final String BOOT_JDBC_PASS = BOOT_DB + ".database.pass";

    public static final String BOOT_DIR = "dir";
    public static final String BOOT_WORKSPACE_DIR = BOOT_DIR + ".workspace";
    public static final String BOOT_TMP_DIR = BOOT_DIR + ".tmp";

    public static final String BOOT_REST = "rest";
    public static final String BOOT_REST_TLS_KEYSTORE_FILE_PATH = BOOT_REST + ".tls.keystore.path";
    public static final String BOOT_REST_TLS_KEYSTORE_PASSWORD = BOOT_REST + ".tls.keystore.password";

    public static final String BOOT_SECRETS = "secrets";
}

File: data/config-schema/src/main/kotlin/net/corda/schema/configuration/BootConfig.kt
package net.corda.schema.configuration

import net.corda.schema.configuration.MessagingConfig.MAX_ALLOWED_MSG_SIZE

/**
 * Configuration paths for values used to bootstrap the worker
 */
object BootConfig {
    const val INSTANCE_ID = "instanceId"
    const val TOPIC_PREFIX = "topicPrefix"
    const val BOOT_MAX_ALLOWED_MSG_SIZE = MAX_ALLOWED_MSG_SIZE

    const val BOOT_KAFKA = "kafka"
    const val BOOT_KAFKA_COMMON = "$BOOT_KAFKA.common"
    const val BOOT_CRYPTO = "crypto"

    const val BOOT_DB = "db"
    const val BOOT_DB_PARAMS = "$BOOT_DB.params"
    const val BOOT_JDBC_URL = "$BOOT_DB_PARAMS.database.jdbc.url"
    const val BOOT_JDBC_USER = "$BOOT_DB_PARAMS.database.user"
    const val BOOT_JDBC_PASS = "$BOOT_DB_PARAMS.database.pass"

    const val BOOT_DIR = "dir"
    const val BOOT_WORKSPACE_DIR = "$BOOT_DIR.workspace"
    const val BOOT_TMP_DIR = "$BOOT_DIR.tmp"
}
